common/component/kafka/kafka.go

```diff
@@ -182,10 +184,28 @@ func (k *Kafka) Init(ctx context.Context, metadata map[string]string) error {
 		// already handled in updateTLSConfig
 	case awsIAMAuthType:
 		k.logger.Info("Configuring AWS IAM authentication")
-		err = updateAWSIAMAuthInfo(k.internalContext, config, meta)
+		region, accessKey, secretKey, assumeRole, sessionName, validateErr := k.ValidateAWS(metadata)
+		if validateErr != nil {
+			return fmt.Errorf("failed to validate AWS IAM authentication fields: %w", validateErr)
+		}
+		opts := awsAuth.Options{
+			Logger:        k.logger,
+			Properties:    metadata,
+			Region:        region,
+			Endpoint:      "",
+			AccessKey:     accessKey,
+			SecretKey:     secretKey,
+			SessionToken:  "",
```

Comment:
why is this field missing, isnt it already part of the auth profile?

Reply:
it was not being utilized before, but i can add it